graph/src/log/elastic.rs

```diff
@@ -272,19 +276,24 @@ impl ElasticDrain {
                         .post(batch_url)
                         .header(CONTENT_TYPE, "application/json"),
                 };
-                header
-                    .body(batch_body)
-                    .send()
-                    .and_then(|response| async { response.error_for_status() })
-                    .map_ok(|_| ())
-                    .unwrap_or_else(move |e| {
+
+                retry("send logs to elasticsearch", &flush_logger)
+                    .limit(max_retries)
+                    .timeout_secs(30)
+                    .run(move || {
+                        header
+                            .try_clone()
+                            .unwrap() // Unwrap: Request body not yet set
```

Comment:
I don't understand this comment.

What's the relation between the `unwrap` and not having the `body` set yet?

I remember one of the times I've used `reqwest` there was some kind of "catch" when unwraping the body, but I think it was on the `response` part. Is this related to it?

I took a quick look through the [docs](https://docs.rs/reqwest/0.11.6/reqwest/) and some old code I've made using the lib and I still don't understand.

Reply:
See https://docs.rs/reqwest/0.11.6/reqwest/struct.RequestBuilder.html#method.try_clone